templates/documents/general_view.html

```
@@ -233,8 +296,11 @@ <h4>{xlt t='Contents'}</h4>
 	</tr>
 	<tr id="DocContents" class="h-100">
 		<td>
-      {if $file->get_mimetype() eq "image/tiff" or $file->get_mimetype() eq "text/plain"}
-			<embed  style="height:84vh; border: 0px" type="{$file->get_mimetype()|attr}" src="{$web_path|attr}as_file=false"></embed>
+      {if $file->get_mimetype() eq "image/tiff"}
+            {* Firefox cannot render TIFF in iframe/embed; use img like core 8.2. *}
+            <img style="max-width:100%; height:auto; max-height:84vh; border:0" alt="{$file->get_name()|attr}" src="{$web_path|attr}as_file=false" />
+            {elseif $file->get_mimetype() eq "text/plain"}
+			<embed  style="height:84vh; border: 0px" type="text/plain" src="{$web_path|attr}as_file=false"></embed>
```

HTMLHint (1.9.2)

[error] Tag must be paired, no start tag: [ </embed> ]

(tag-pair)